### PR TITLE
Relations endpoint support

### PR DESCRIPTION
**Pull request by bwindels**
Needed for https://github.com/vector-im/riot-web/issues/9490
React-sdk PR: https://github.com/matrix-org/matrix-react-sdk/pull/3144

### CODE REVIEW

File: src/base-apis.js [hunk 1]
```diff
@@ -439,6 +439,27 @@ MatrixBaseApis.prototype.createRoom = function(options, callback) {
     );
 };
 
+MatrixBaseApis.prototype.fetchRelations =
```
Comment (jryans) [inline src/base-apis.js]: Please add function docs like other functions here.

File: src/base-apis.js [hunk 2]
```diff
@@ -439,6 +439,27 @@ MatrixBaseApis.prototype.createRoom = function(options, callback) {
...
+    async function(roomId, eventId, relationType, eventType, opts) {
+    const queryParams = {};
+    if (opts.token) {
+        queryParams.from = opts.token;
```
Comment (jryans) [inline src/base-apis.js]: I assume the name `token` matches other APIs?

File: src/base-apis.js [hunk 3]
```diff
@@ -439,6 +439,27 @@ MatrixBaseApis.prototype.createRoom = function(options, callback) {
...
+    }
+    const queryString = utils.encodeParams(queryParams);
+    const path = utils.encodeUri(
+        "/rooms/$roomId/relations/$eventId/$relationType/$eventType?" + queryString, {
+            $roomId: roomId,
+            $eventId: eventId,
+            $relationType: relationType,
+            $eventType: eventType,
+        });
+    // valid options include: room_alias_name, visibility, invite
```
Comment (jryans) [inline src/base-apis.js]: What does this means? Options for what...?

File: src/client.js [hunk 4]
```diff
@@ -3965,6 +3965,20 @@ MatrixClient.prototype.getCanResetTimelineCallback = function() {
     return this._canResetTimelineCallback;
 };
 
+MatrixClient.prototype.relations =
```
Comment (jryans) [inline src/client.js]: This should also have function docs like others here.

**Comment by bwindels** [comment]
Sorry for the messy initial state, should be better now.

**Pull request merged**